### ISSUE TITLE
Idea/suggestion: add templates from Avery.nl labels

### ISSUE THREAD

**Issue by denilsonsa** [johnbeard/inkscape-label-guides#7]
**TL;DR:** I extracted data for about 75 labels from the Avery company. It would be nice to add it to your extension, as I believe it can be helpful to other people as well.

---

Avery is an European brand for labels. They have [many sizes](https://www.avery.nl/etiketten/vormen), and they have their own [Design & Print Online](https://www.avery.nl/software/avery-design-print) web app. Their web app supports all of their labels.

If you use the web app and go further until the step 4, you will notice the page preview is rendered using an SVG element, and thus we could theoretically extract the SVG from that page. That, however, takes a lot of work (there are too many layouts, and we would need to interact with the webpage to extract data).

Instead, I went to other pages from their website and I found they provide DOCX templates for many of their labels. I couldn't find as many templates as in their official design web app, but I found about almost 80 templates, and they cover the most common label sizes and shapes.

To make it easier, and to make it ready to be inserted into your Inkscape extension (or into any other tool), I wrote some Python code to automatically extract data from all of the templates I could find. All the data is available at <https://gist.github.com/denilsonsa/9ab0f7ab30f4e56e233825bddc06f65a>. Feel free to use it to add to your extension. I hope it helps other people as well.

**Comment by johnbeard** [comment]
Hi @denilsonsa thanks for the data.

Do you know if Avery labels are always dead-centre in the page. There is no indication of what the margins are for these labels.

Thanks!

**Comment by denilsonsa** [comment]
Well, I'm assuming they are. The one I had to print last week was completely centered.

I don't know how to extract that data (margins) from the DOCX files. (That was my first and only time I ever used the `python-docx` module.)

I've updated the code from https://gist.github.com/denilsonsa/9ab0f7ab30f4e56e233825bddc06f65a to also save all the DOCX files and use `libreoffice` to convert them to PDFs. You may want to use it to extract some data from the PDF, or further convert the PDFs to a high-res raster image (so it would be possible to measure the amount of blank rows and cols of pixels, and thus measure an approximation of the margin), or even to compare some of those PDFs to the SVG output from this extension.

Since there are 77 files, it's hard to validate them all manually, so we need to either check them by code, or pick a small sample of them for manual inspection (which is probably fine).

Another idea is to mirror the PDFs horizontally and/or vertically and compare them with the original version. Both versions should line up if they are aligned to the center. That could be done by code, possibly using ImageMagick to convert them to raster images and doing some basic manipulation.

**Comment by johnbeard** [comment]
Hi @denilsonsa, yeah I checked a few manually and they all look centred. I asked both Avery UK and US and neither had the information to hand without escalating on a per-product basic. Odd, that they don't just have it, but there you go.

Dissecting the DOCX is probably doable, but I don't know if it's worth it - it only matters for the ones in your list but not in the extension (many are already included), and it's not too hard to just open the template. Maybe I'll just do that one day when I'm bored!

Side note, I did notice that the industrial PCL range does have good templates with margins, but they are _not_ centred!

![image](https://github.com/user-attachments/assets/1a4dea10-a203-4f0b-af8e-749be37fa127)
